### PR TITLE
add support for trace-dir option

### PR DESCRIPTION
**Pull request by SunilKumarKori**
eal trace-dir option is added which provide flexibility to configure
directory where trace output should be generated.

Signed-off-by: Sunil Kumar Kori <skori@marvell.com>

### CODE REVIEW

File: doc/guides/linux_gsg/eal_args.include.rst [hunk 1]
```diff
@@ -155,6 +155,17 @@ Debugging options
 
     Can be specified multiple times up to 32 times.
 
+*   ``--trace-dir=<directory path>``
+
+    Specify trace directory for trace output. For example:
+
+    Configuring ``/usr/`` as a trace output directory::
```
Comment (jerinjacobk) [inline doc/guides/linux_gsg/eal_args.include.rst]: give example as /tmp/

File: lib/librte_eal/common/eal_common_trace_utils.c [hunk 2]
```diff
@@ -255,21 +255,60 @@ trace_level_args_apply(const char *arg)
...
+		dir_path = (char *)calloc(1, sizeof(trace->dir));
+		if (dir_path == NULL) {
+			trace_err("fail to allocate memory\n");
+			return -ENOMEM;
+		}
 
-	rc = trace_dir_update("dpdk-trace/");
-	if (rc < 0)
-		return rc;
+		home_dir = getenv("HOME");
```
Comment (jerinjacobk) [inline lib/librte_eal/common/eal_common_trace_utils.c]: Move to figure out the home directory to separate function.

Comment (jerinjacobk) [inline lib/librte_eal/common/eal_common_trace_utils.c]: getenv("HOME") may not portable as the code needs to work for freebsd. See lttng code to see how they
figure out the default.

https://github.com/jerinjacobk/lttng-overhead.git has step to compile and and run lttnng.

Comment (SunilKumarKori) [reply lib/librte_eal/common/eal_common_trace_utils.c]: I have verified that getenv("HOME") is applicable for freebsd also.

File: lib/librte_eal/common/eal_common_trace_utils.c [hunk 3]
```diff
@@ -255,21 +255,66 @@ trace_level_args_apply(const char *arg)
...
+		if (dir_path == NULL) {
+			trace_err("fail to allocate memory\n");
+			return -ENOMEM;
+		}
 
-	rc = trace_dir_update("dpdk-trace/");
-	if (rc < 0)
-		return rc;
+		home_dir = trace_dir_default_path_get();
+		sprintf(dir_path, "%s/", home_dir);
```
Comment (jerinjacobk) [inline lib/librte_eal/common/eal_common_trace_utils.c]: This line can move to trace_dir_default_path_get() and check the error as well.

**Pull request merged**